grafana-plugin/src/pages/incident/Incident.tsx

```diff
@@ -89,6 +93,12 @@ class IncidentPage extends React.Component<IncidentPageProps, IncidentPageState>
     store.alertGroupStore.updateSilenceOptions();
   }
 
+  componentWillUnmount(): void {
```

Comment:
Wouldn't it be better to make a generic React component that handles this cleanup so that we end with less duplicated lines in the affected components?
Like a `BaseContext<P, S>` component that extends `React.Component<P, S>` where you add the `contextType`, `context`, `cleanup` function and `componentWillUnmount` (and if ever we're going to add more to the componentWillUnmount for the extended component we can just  call `cleanup` on the base class instead)